arangod/Agency/Agent.cpp

```diff
@@ -313,15 +313,27 @@ void Agent::reportIn(std::string const& peerId, index_t index, size_t toLog) {
 }
 
 /// @brief Report a failed append entry call from AgentCallback
-void Agent::reportFailed(std::string const& slaveId, size_t toLog) {
+void Agent::reportFailed(std::string const& slaveId, size_t toLog, bool sent) {
   if (toLog > 0) {
     // This is only used for non-empty appendEntriesRPC calls. If such calls
     // fail, we have to set this earliestPackage time to now such that the
-    // main thread tries again immediately:
+    // main thread tries again immediately: and for that agent starting at 0
+    // which effectively will be _state.firstIndex().
     MUTEX_LOCKER(guard, _tiLock);
     LOG_TOPIC(DEBUG, Logger::AGENCY)
       << "Resetting _earliestPackage to now for id " << slaveId;
     _earliestPackage[slaveId] = steady_clock::now();
```

Comment:
I am concerned that blindly setting _earliestPackage to now() can create a fast loop if there is something really wrong with the follower.  First time through the failed, now() is ok.  Not subsequent retries.  Basically saying that there should be a parallel _lastFailed to _lastAcked.  _lastFailed gets set to zero when _lastAcked set to a true value.  Without this, another 100% CPU condition is still going to crop up ... we just do not know the flavor of the next bug.

Reply:
Further suggest that _lastFailed be a count, not time.  You can use the count to create incremental retry pacing if desired.  But more importantly you can use the count to more easily decide when to start raising hell in the logs and how often.

Reply:
Let's not change the format of this for this iteration but open for discussion. 
It cannot create 100%: 
 * cause if timeout on other side, time passes until we come back here. 
 * if error still milliseconds pass. 
I am reluctant to changes in agents on short notice.